teslajsonpy/vehicle.py

```diff
@@ -2,18 +2,22 @@ class VehicleDevice:
     def __init__(self, data, controller):
         self._id = data['id']
         self._vehicle_id = data['vehicle_id']
+        self._display_name = data['display_name']
         self._vin = data['vin']
         self._state = data['state']
         self._controller = controller
         self.should_poll = True
 
     def _name(self):
-        return 'Tesla Model {} {}'.format(
-            str(self._vin[3]).upper(), self.type)
+        return ('{} {}'.format(self._display_name, self.type) if
+                self._display_name is not None and
+                self._display_name != self._vin[-6:]
+                else 'Tesla Model {} {}'.format(str(self._vin[3]).upper(),
+                                                self.type))
```

Comment:
Maybe it is better to calculate some hash of it? As it is really bad idea to disclose this item.

Reply:
I think it's useful for a user to be able to match the ID to some value they know since some will have multiple vehicles. I think we could truncate it further so it's not the full serial number.  I doubt people will see a collision at 3-4 digits.  
Here's a vin decoding link for [reference](https://www.autocheck.com/vehiclehistory/autocheck/en/vinbasics).

Reply:
Because this is the name that will display in Home Assistant (and therefore generate the entity_id), I think you should remove the VIN from this entirely. The end result should simply be "Tesla Model X Door Locks" or similar, if there is no `_display_name`. The VIN will be used for the `unique_id` and the user will be able to change the display name to whatever they want.

This suggestion would solve the problem that zabuldon highlights above of not exposing the VIN.

Reply:
@OverloadUT While it may not be a sizable population, there are users with multiple vehicles where that would result in a name collision, particularly as the Model 3 becomes more affordable. I'm inclined to just do the last 3 chars of the VIN for the unique ID and for the name if none have been set.

Reply:
@alandtse @OverloadUT Guys, what do you think if we still use naming on "Library" side, but allow users to configure names of Tesla's on HA side? I saw this behavior with insteon_local components. They allow you to rename thing when you set it up.

Reply:
@zabuldon Exposing a unique_id to Home Assistant gives the user all of the power. They can change the display names and entity_ids of everything that has a unique_id at runtime without needing to even restart. So we get that for free :)